runelite-client/src/main/java/net/runelite/client/plugins/clanchat/ClanChatPlugin.java

```diff
@@ -70,6 +106,50 @@ public void updateClanChatTitle()
 		}
 	}
 
+	private boolean isClanChatClosed()
+	{
+		Widget widget = client.getWidget(WidgetInfo.CLAN_CHAT);
+
+		return widget == null || widget.isHidden();
+	}
+
+	@Override
+	public boolean onChatboxInput(ChatboxInput chatboxInput)
+	{
+		if (!config.swapChat() && (!config.mobileChat() || isClanChatClosed()))
+		{
+			return false;
+		}
+
+		String msg = chatboxInput.getValue();
+
+		if (msg.startsWith("/"))
+		{
+			msg = msg.substring(1);
+			chatboxInput.setChatType(0);
```

Comment:
I believe we already have the chat types defined somewhere so maybe you should use them instead of 0/2

Reply:
Sending chat types are different than the receiving chat types.

Reply:
Yeah so probably make some enum/constants for them.